src/test/java/duke/services/Parser_Test.java

```diff
@@ -0,0 +1,58 @@
+package duke.services;
+
+import duke.tasks.Todo;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class Parser_Test {
+    @Test
+    public void testGetTaskNumber() {
+        //exception on trying to get task number when there are no tasks
+        assertThrows(IllegalArgumentException.class, () -> Parser.getTaskNumber(new String[] {"mark", "1"}));
+        TaskList.addTask(new Todo(""));
+        assertEquals(Parser.getTaskNumber(new String[] {"mark", "1"}), 1);
+        assertThrows(IllegalArgumentException.class, () -> Parser.getTaskNumber(new String[] {"mark"}));
+        assertThrows(IllegalArgumentException.class, () -> Parser.getTaskNumber(new String[] {"mark", "2a"}));
+        assertThrows(IllegalArgumentException.class, () -> Parser.getTaskNumber(new String[] {"mark", "9"}));
+    }
+
+    @Test
+    public void testGetDescriptionAndDate() {
+        String[] event1 = new String[] {"event", "presentation", "/at", "9/10/2022"};
+        assertEquals(Parser.getDescription(event1, "/at"), "presentation");
+        assertEquals(Parser.getTiming(event1, "/at"), "9/10/2022");
+        String[] deadline1 = new String[] {"deadline", "do es2660", "", "", "essay", "/by", "tmr", "night",
+                "", "", "", "or", "ask", "4", "extension"};
+        assertEquals(Parser.getDescription(deadline1, "/by"), "do es2660   essay");
+        assertEquals(Parser.getTiming(deadline1, "/by"), "tmr night    or ask 4 extension");
+        //does it throw exception for missing all other arguments?
+        assertThrows(IllegalArgumentException.class,
+                () -> Parser.getDescription(new String[] {"event"}, "/at"));
+        //exception on timing flag included but missing desc?
+        assertThrows(IllegalArgumentException.class,
+                () -> Parser.getDescription(new String[] {"deadline", "/by"}, "/by"));
+        //exception on missing flag?
+        assertThrows(IllegalArgumentException.class,
+                () -> {
+                    String[] deadline = new String[]{"deadline", "help", "by"};
+                    Parser.getDescription(deadline, "/by");
+                    Parser.getTiming(deadline, "/by");
+                });
```

Comment:
I think it would better to have each of these asserts be separate test cases. 
:sweat_smile: